openassessment/workflow/models.py

```diff
@@ -309,17 +312,20 @@ def update_from_assessments(self, assessment_requirements):
 
         new_staff_score = self.get_score(assessment_requirements, {'staff': step_for_name.get('staff', None)})
         if new_staff_score:
-            old_score = self.score
-            if not old_score or old_score['points_earned'] != new_staff_score['points_earned']:
+            old_score = sub_api.get_latest_score_for_submission(self.submission_uuid)
+            if not old_score or not old_score.get('staff_id') or old_score['points_earned'] != new_staff_score['points_earned']:
```

Comment:
This is kind of complicated, and I would suggest explaining what each clause is testing for (old score is a staff score, etc.)